claripy/backends/backend_smtlib_solvers/cvc4_popen.py

```diff
@@ -0,0 +1,57 @@
+import subprocess
+import logging
+
+import re
+from . import SMTLibSolverBackend, PopenSolverProxy
+
+log = logging.getLogger(__name__)
+
+def get_version():
+    try:
+        version_string = subprocess.check_output(['cvc4', '--version']).decode('utf-8')
+        version_match = re.match('This is CVC4 version (.*)\n', version_string)
+
+        if not version_match:
+            return False, None, "Found malformed version string: {}".format(version_string)
+
+        return True, version_match.group(1), None
+
+    except subprocess.CalledProcessError as ex:
+        return False, None, "Not found, error: {}".format(ex)
+    except OSError as ex:
+        return False, None, "Not found, error: {}".format(ex)
+
+
+IS_INSTALLED, VERSION, ERROR = get_version()
+
+if IS_INSTALLED:
+    class CVC4Proxy(PopenSolverProxy):
```

Comment:
Can we make these classes not optional? In other words, we should always create these classes regardless of if CVC4 (or ABC) is installed or not. We should only check if the underlying solver is installed or not when we actually use them (in create_process(), for example).

Reply:
Uhm I don't see the benefit of it, actually, it would be more complicated in my opinion. If I add the if during the `create_process` function I need to throw an exception anyway it the solver is not installed.

Reply:
>  I need to throw an exception anyway it the solver is not installed

Of course. IMO having the existence of a class decided by some variable is non-standard and counter-intuitive. We should always be able to reference this class in other part of the code regardless of whether CVC4 (or whatever other solvers) is installed or not. In addition, we should add an `.installed` static property to the `CVC4Proxy` class so users and programmers can easily tell if CVC4 is currently installed or not by checking `CVC4Proxy.installed`.